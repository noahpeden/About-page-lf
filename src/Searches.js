import React from 'react';

export default props => {
	const { history } = props;
	return (
		<div>
			{' '}{/* Render the list of searches */
			history.map(search =>
				<li key={search.id}>
					{search.text}
				</li>
			)}
		</div>
	);
};
