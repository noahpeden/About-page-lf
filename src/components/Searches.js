import React, { Component } from 'react';
import fire from '../firebase';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

export default class Searches extends Component {
	state = {
		searches: []
	};
	history = [];

	componentWillMount() {
		let searchesRef = fire
			.database()
			.ref('search')
			.orderByKey()
			.limitToLast(100);
		searchesRef.on('child_added', snapshot => {
			let searches = snapshot.val();
			this.history.push({
				id: snapshot.key,
				currently: searches.currently,
				timeStamp: searches.timeStamp,
				description: searches.description,
				cityName: searches.cityName
      });
      
      this.setState({ searches: this.history });
		});
	}

	render() {
		const Wrapper = styled.div`
			margin-top: 30px;
			color: white;
			font-size: 20px;
			font-family: sans-serif;
			text-align: center;
			ul {
				margin: 10px;
      }
      .home {
        position: absolute;
        top: 0;
        left: 0;
      }
      .home:hover {
        text-decoration: underline;
      }
		`;
		return (
			<Wrapper>
				<Link style={{ textDecoration: 'none', color: 'white' }} to="/">
					<div className="home">HOME</div>
				</Link>
				<div className="searches">
					<h1>PAST SEARCHES</h1>
					{/* Render the list of searches */
					this.state.searches.map(search =>
						<ul key={search.id}>
							<li>
								{search.cityName}
							</li>
							<li>
								{search.currently}
							</li>
							<li>
								{search.timeStamp}
							</li>
							<li>
								{search.description}
							</li>
						</ul>
					)}
				</div>
			</Wrapper>
		);
	}
}
