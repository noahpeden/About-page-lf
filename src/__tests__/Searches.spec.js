import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Searches from '../components/Searches';

test('ClientApp renders correctly', () => {
	const component = shallow(<Searches />);
	expect(component).toMatchSnapshot();
});
