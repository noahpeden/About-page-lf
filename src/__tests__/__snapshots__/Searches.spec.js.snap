// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientApp renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <Link
        replace={false}
        style={
            Object {
                "color": "white",
                "textDecoration": "none",
              }
        }
        to="/"
    >
        <div
            className="home"
        >
            HOME
        </div>
    </Link>
    <div
        className="searches"
    >
        <h1>
            PAST SEARCHES
        </h1>
    </div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <Link
            replace={false}
            style={
                  Object {
                        "color": "white",
                        "textDecoration": "none",
                      }
            }
            to="/"
      >
            <div
                  className="home"
            >
                  HOME
            </div>
      </Link>
      <div
            className="searches"
      >
            <h1>
                  PAST SEARCHES
            </h1>
      </div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Searches />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Searches {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "history": Array [],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "searches": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <Link
                    replace={false}
                    style={
                              Object {
                                        "color": "white",
                                        "textDecoration": "none",
                                      }
                    }
                    to="/"
          >
                    <div
                              className="home"
                    >
                              HOME
                    </div>
          </Link>
          <div
                    className="searches"
          >
                    <h1>
                              PAST SEARCHES
                    </h1>
          </div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <Link
                    replace={false}
                    style={
                              Object {
                                        "color": "white",
                                        "textDecoration": "none",
                                      }
                    }
                    to="/"
          >
                    <div
                              className="home"
                    >
                              HOME
                    </div>
          </Link>
          <div
                    className="searches"
          >
                    <h1>
                              PAST SEARCHES
                    </h1>
          </div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Searches />,
}
`;
